internal/subshell/pwsh/pwsh.go

```diff
@@ -0,0 +1,169 @@
+package pwsh
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/osutils"
+	"github.com/ActiveState/cli/internal/osutils/user"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/subshell/cmd"
+	"github.com/ActiveState/cli/internal/subshell/sscommon"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+var escaper *osutils.ShellEscape
+
+func init() {
+	escaper = osutils.NewBatchEscaper()
+}
+
+// SubShell covers the subshell.SubShell interface, reference that for documentation
+type SubShell struct {
+	binary string
+	rcFile *os.File
+	cmd    *exec.Cmd
+	env    map[string]string
+	errs   chan error
+}
+
+const Name string = "powershell"
+
+// Shell - see subshell.SubShell
+func (v *SubShell) Shell() string {
+	return Name
+}
+
+// Binary - see subshell.SubShell
+func (v *SubShell) Binary() string {
+	return v.binary
+}
+
+// SetBinary - see subshell.SubShell
+func (v *SubShell) SetBinary(binary string) {
+	v.binary = binary
+}
+
+// WriteUserEnv - see subshell.SubShell
+func (v *SubShell) WriteUserEnv(cfg sscommon.Configurable, env map[string]string, envType sscommon.RcIdentification, userScope bool) error {
+	cmdShell := &cmd.SubShell{}
+	if err := cmdShell.WriteUserEnv(cfg, env, envType, userScope); err != nil {
+		return errs.Wrap(err, "Forwarded WriteUserEnv call failed")
+	}
+
+	return nil
+}
+
+func (v *SubShell) CleanUserEnv(cfg sscommon.Configurable, envType sscommon.RcIdentification, userScope bool) error {
+	cmdShell := &cmd.SubShell{}
+	if err := cmdShell.CleanUserEnv(cfg, envType, userScope); err != nil {
+		return errs.Wrap(err, "Forwarded CleanUserEnv call failed")
+	}
+
+	return nil
+}
+
+func (v *SubShell) RemoveLegacyInstallPath(_ sscommon.Configurable) error {
+	return nil
+}
+
+func (v *SubShell) WriteCompletionScript(completionScript string) error {
+	return locale.NewError("err_writecompletions_notsupported", "{{.V0}} does not support completions.", v.Shell())
+}
+
+func (v *SubShell) RcFile() (string, error) {
+	home, err := user.HomeDir()
+	if err != nil {
+		return "", errs.Wrap(err, "Could not get home dir")
+	}
+
+	return filepath.Join(home, "Documents", "WindowsPowerShell", "Microsoft.PowerShell_profile.ps1"), nil
```

Comment:
Eww, littering the user's documents directory. I'm guessing PowerShell writes to this directory if needed? (It doesn't exist for me by default.) And are we sure this is the exact path? (e.g. no space between "Windows" and "PowerShell"?)

Reply:
This is powershell's equivalent of a bashrc file. 

https://activestatef.atlassian.net/browse/DX-2944

Powershell reads this file if it exists. It doesn't write to it afaik.